src/connection/host_lookup.c

```diff
@@ -0,0 +1,457 @@
+	{
+//err_msg : convert binary to text fail
+	  return NULL;
+	}
+
+    }
+
+/*Look up in the hostent_List*/
+  for (i = 0; i < host_conf_element_Count; i++)
+    {
+      if ((lookup_case == HOSTNAME_TO_IPADDR) && (strcmp (hostname, hostent_List[i].hostname) == 0))
+	{
+	  find_index = i;
+	  break;
+	}
+
+      if ((lookup_case == IPADDR_TO_HOSTNAME) && (strcmp (addr_transform_buf, hostent_List[i].ipaddr) == 0))
+	{
+	  find_index = i;
+	  break;
+	}
+    }
+
+  if (find_index < 0)
+    {
+      return NULL;
+//err_msg : hosts.conf has no info
+    }
+
+  if (inet_pton (AF_INET, hostent_List[find_index].ipaddr, addr_transform_buf2) < 1)
+    {
+//err_set : convert binary to text fail -> reverse
+      return NULL;
+    }
+
+  /*set the hostent struct for the return value */
+  hp.h_addrtype = AF_INET;
+  hp.h_length = 4;
+  hp.h_name = (char *) malloc (sizeof (char) * (HOSTNAME_BUF_SIZE + 1));
+  hp.h_aliases = (char **) malloc (sizeof (char *));
+  hp.h_aliases[0] = (char *) malloc (sizeof (char) * (HOSTNAME_BUF_SIZE + 1));
+  hp.h_addr_list = (char **) malloc (sizeof (char *) * HOSTNAME_BUF_SIZE);
+  hp.h_addr_list[0] = (char *) malloc (sizeof (char) * IPADDR_LEN);
+
+  strcpy (hp.h_name, hostent_List[find_index].hostname);
+  strcpy (hp.h_aliases[0], hostent_List[find_index].hostname);
+  memcpy (hp.h_addr_list[0], addr_transform_buf2, sizeof (hp.h_addr_list[0]));
+
+  return &hp;
+}
+
+static int
+host_conf_load ()		//set hash table to discover error
+{
+  FILE *fp;
+  char file_line[HOSTNAME_BUF_SIZE + 1];
+  char line_buf[HOSTNAME_BUF_SIZE + 1];
+  char host_conf_file_full_path[PATH_MAX];
+  char *hosts_conf_dir;
+
+  char *token;
+  char *save_ptr_strtok;
+  /*delimiter */
+  char *delim = " \t\n";
+
+  int host_count = 0;
+
+  /*False, if hostent_List[index].hostname is set */
+  bool hostent_flag;
+  HOSTENT_INSERT_TYPE hostent_insert_Type;
+
+  memset (file_line, 0, HOSTNAME_BUF_SIZE + 1);
+  memset (line_buf, 0, HOSTNAME_BUF_SIZE + 1);
+
+  hosts_conf_dir = envvar_confdir_file (host_conf_file_full_path, PATH_MAX, "hosts.conf");
+  fp = fopen (hosts_conf_dir, "r");
+  if (fp == NULL)
+    {
+//err_msg : file open fail
+//er_set_with_oserror (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_IO_MOUNT_FAIL, 1, host_conf_file_full_path);
+//fprintf (stdout, msgcat_message (MSGCAT_CATALOG_CUBRID, MSGCAT_SET_ERROR, -ER_IO_MOUNT_FAIL));
+      return -1;
+    }
+
+  while (fgets (file_line, HOSTNAME_BUF_SIZE + 1, fp) != NULL)
+    {
+//verifying duplicated hostname will be implemented soon
+      if (file_line[0] == '#')
+	continue;
+      if (file_line[0] == '\n')
+	continue;
+
+      token = strtok_r (file_line, delim, &save_ptr_strtok);
+      hostent_flag = INSERT_HOSTNAME;
+
+      char map_ipaddr[IPADDR_LEN];
+      char map_hostname[HOSTNAME_BUF_SIZE + 1];
+
+      do
+	{
+	  if (*token == '\0' || *token == '#')
+	    {
+	      break;
+	    }
+	  if (hostent_flag == INSERT_HOSTNAME)
+	    {
+	      if (sizeof (token) > HOSTNAME_BUF_SIZE + 1)
```

Comment:
To get the string length of token, you must use strlen.

Reply:
Right, I will fix it!